Don't stomp on the output BIGNUM -- it may be one of our operands.
--- openssl-0.9.7a/crypto/engine/hw_ubsec.c	2003-06-10 18:54:24.000000000 -0400
+++ openssl-0.9.7a/crypto/engine/hw_ubsec.c	2003-06-10 18:54:13.000000000 -0400
@@ -561,7 +561,6 @@
 		UBSECerr(UBSEC_F_UBSEC_MOD_EXP, UBSEC_R_BN_EXPAND_FAIL);
 		return 0;
 	}
-	memset(r->d, 0, BN_num_bytes(m));
 
 	if ((fd = p_UBSEC_ubsec_open(UBSEC_KEY_DEVICE_NAME)) <= 0) {
 		fd = 0;
